### PR TITLE
Add basic type definitions for issueType api

### PR DESCRIPTION
**Pull request by G-Rath**
🎉

### CODE REVIEW

File: api/issueType.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ function IssueTypeClient(jiraClient) {
      *
      * @method getAllIssueTypes
      * @memberOf IssueTypeClient#
-     * @param opts Ignored
+     * @param {Object} [opts] Ignored
      * @param {Function} [callback] Called when the issue types have been retrieved.
```
Comment (MrRefactoring) [inline api/issueType.js]: If you can, replace `{Function}` -> `{callback}` 😀

**Comment by MrRefactoring** [comment]
You forgot to add issueType to index.d.ts 🙂

**Comment by G-Rath** [comment]
> You forgot to add issueType to index.d.ts 🙂

Good catch!

Plus you've reminded me I should use `export =` in `issueType.d.ts` :D

**Comment by MrRefactoring** [comment]
Can you give a comment about why we use `export =`, not `export declare class ...`?

**Comment by G-Rath** [comment]
Sure!

We use it b/c that represents the export in the `js`; you can see this in the playground, but in a nutshell, here's what each type of export transpiles to:

-------

```
export default 'hello';
```

Becomes
```
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = 'hello';
```

-------

```
export const hello = 'hello';
```

Becomes
```
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.hello = 'hello';
```

------

```
export = 'hello';
```

Becomes
```
"use strict";
module.exports = 'hello';
```

------

Since the exports are being done using `module.exports`, the correct way to represent that in TypeScript is to use `export = ...` 🙂

**Comment by MrRefactoring** [comment]
This is new knowledge for me, thanks!

**Comment by G-Rath** [comment]
No problem! It's a bit of an edge-case in TS land, that you generally don't need to worry about unless you're writing types manually 😬

**Comment by MrRefactoring** [comment]
Thanks! 🤪

**Pull request merged**